Doc/reference/lexical_analysis.rst

```diff
@@ -351,6 +351,25 @@ exactly as written here:
    assert     del        global     not        with
    async      elif       if         or         yield
 
+
+.. _soft-keywords:
+
+Soft Keywords
+-------------
+
+.. index:: soft keyword, keyword
+
+.. versionadded:: 3.10
+
+The identifiers ``match``, ``case`` and ``_`` can syntactically act as keywords in some
+specific contexts related to the pattern matching statement, but this distinction is done
+at the parser level, not when tokenizing.
```

Comment:
Extremely minor nit; feel free to ignore if you disagree. I think there should be at least a sentence describing what a _soft keyword_ actually is, similar to the keyword section above.

```suggestion
Some identifiers are only reserved under specific contexts. These are known as
*soft keywords*.  The identifiers ``match``, ``case`` and ``_`` can
syntactically act as keywords in contexts related to the pattern matching
statement, but this distinction is done at the parser level, not when
tokenizing.
```

Reply:
Nice phrasing, thanks!